jgrapht-core/src/test/java/org/jgrapht/AStarShortestPathTest.java

```diff
@@ -0,0 +1,162 @@
+    };
+
+    private WeightedGraph<Node, DefaultWeightedEdge> graph;
+    private Node sourceNode;
+    private Node targetNode;
+
+    private void readLabyrinth(String[] labyrinth){
+        graph=new SimpleWeightedGraph<Node, DefaultWeightedEdge>(DefaultWeightedEdge.class);
+
+        //Create the nodes
+        Node[][] nodes=new Node[labyrinth.length][labyrinth[0].length()];
+        for(int i=0; i<labyrinth.length; i++){
+            for(int j=0; j<labyrinth[0].length(); j++){
+                if(labyrinth[i].charAt(j)=='#' || labyrinth[i].charAt(j)==' ')
+                    continue;
+                nodes[i][j]=new Node(i,j);
+                graph.addVertex(nodes[i][j]);
+                if(labyrinth[i].charAt(j)=='S')
+                    sourceNode=nodes[i][j];
+                else if(labyrinth[i].charAt(j)=='T')
+                    targetNode=nodes[i][j];
+            }
+        }
+        //Create the edges
+        //a. Horizontal edges
+        for(int i=0; i<labyrinth.length; i++) {
+            for (int j = 0; j < labyrinth[0].length()-2; j++) {
+                if(nodes[i][j] == null || nodes[i][j+2]==null)
+                    continue;
+                Graphs.addEdge(graph, nodes[i][j], nodes[i][j+2], 1);
+            }
+        }
+        //b. Vertical edges
+        for(int i=0; i<labyrinth.length-1; i++) {
+            for (int j = 0; j < labyrinth[0].length(); j++) {
+                if(nodes[i][j] == null || nodes[i+1][j]==null)
+                    continue;
+                Graphs.addEdge(graph, nodes[i][j], nodes[i+1][j], 1);
+            }
+        }
+    }
+
+    public void testLabyrinth1(){
+        this.readLabyrinth(labyrinth1);
+        AStarShortestPath<Node, DefaultWeightedEdge> aStarShortestPath=new AStarShortestPath<Node, DefaultWeightedEdge>(graph);
+        GraphPath<Node, DefaultWeightedEdge> path=aStarShortestPath.getShortestPath(sourceNode, targetNode, new ManhattanDistance());
+        assertNotNull(path);
+        assertEquals((int)path.getWeight(), 47);
+        assertEquals(path.getEdgeList().size(), 47);
+        assertEquals(Graphs.getPathVertexList(path).size(), 48);
+
+        path=aStarShortestPath.getShortestPath(sourceNode, targetNode, new EuclideanDistance());
+        assertNotNull(path);
+        assertEquals((int)path.getWeight(), 47);
+        assertEquals(path.getEdgeList().size(), 47);
+    }
+
+    public void testLabyrinth2(){
+        this.readLabyrinth(labyrinth2);
+        AStarShortestPath<Node, DefaultWeightedEdge> aStarShortestPath=new AStarShortestPath<Node, DefaultWeightedEdge>(graph);
+        GraphPath<Node, DefaultWeightedEdge> path=aStarShortestPath.getShortestPath(sourceNode, targetNode, new ManhattanDistance());
+        assertNull(path);
+    }
+
+    public void testMultiGraph(){
+        WeightedMultigraph<Node, DefaultWeightedEdge> multigraph=new WeightedMultigraph<Node, DefaultWeightedEdge>(DefaultWeightedEdge.class);
+        Node n1=new Node(0,0);
```

Comment:
I've added some comments to the test class to clarify this. The test you are referring to is there to test whether multigraphs are processed correctly. The previous A* pull requests couldn't handle MultiGraphs correctly, so I added a test to ensure that future improvements/changes to the A* code wouldn't break this feature. The coordinates and the ManhattanDistance AStarAdmissibleHeuristic are irrelevant in this test.